libindy/src/services/metrics/mod.rs

```diff
@@ -125,10 +182,58 @@ mod test {
 
         metrics_service.append_command_metrics(&mut metrics_map);
 
-        assert_eq!(metrics_map.len(), COMMANDS_COUNT * 4);
-        assert!(metrics_map.contains_key("issuer_command_create_schema_queued_commands_count"));
-        assert!(metrics_map.contains_key("issuer_command_create_schema_queued_commands_duration_ms"));
-        assert!(metrics_map.contains_key("issuer_command_create_schema_executed_commands_count"));
-        assert!(metrics_map.contains_key("issuer_command_create_schema_executed_commands_duration_ms"));
+        assert!(metrics_map.contains_key("commands_count"));
+        assert!(metrics_map.contains_key("commands_duration_ms"));
+        assert_eq!(
+            metrics_map
+                .get("commands_count")
+                .unwrap()
+                .as_array()
+                .unwrap()
+                .len(),
+            COMMANDS_COUNT * 2
+        );
+        assert_eq!(
+            metrics_map
+                .get("commands_duration_ms")
+                .unwrap()
+                .as_array()
+                .unwrap()
+                .len(),
+            COMMANDS_COUNT * 2
+        );
+
+        let commands_count = metrics_map
+            .get("commands_count")
+            .unwrap()
+            .as_array()
+            .unwrap();
+        let commands_duration_ms = metrics_map
+            .get("commands_duration_ms")
+            .unwrap()
+            .as_array()
+            .unwrap();
+
+        let expected_commands_count = [
+            json!({"tags":{"command":"payments_command_build_set_txn_fees_req_ack","stage":"executed"},"value":0}),
+            json!({"tags":{"command":"metrics_command_collect_metrics","stage":"queued"},"value":0}),
+            json!({"tags":{"command":"cache_command_purge_cred_def_cache","stage":"executed"},"value":0}),
+            json!({"tags":{"command": "non_secrets_command_fetch_search_next_records","stage":"queued"},"value":0}),
+        ];
+
+        let expected_commands_duration_ms = [
+            json!({"tags":{"command":"payments_command_build_set_txn_fees_req_ack","stage":"executed"},"value":0}),
+            json!({"tags":{"command":"metrics_command_collect_metrics","stage":"queued"},"value":0}),
+            json!({"tags":{"command":"cache_command_purge_cred_def_cache","stage":"executed"},"value":0}),
+            json!({"tags":{"command":"non_secrets_command_fetch_search_next_records","stage":"queued"},"value":0}),
+        ];
+
+        for command in &expected_commands_count {
+            assert!(commands_count.contains(&command));
```

Comment:
I'd prefer to avoid loops in tests for a lot of reasons, but the most compelling one in this case is that when this assert fails you won't see why exactly until you rerun this under debugger.